File: src/main/java/com/procurement/notice/model/budget/FsPlanning.java
package com.procurement.notice.model.budget;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.Getter;

@Getter
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "budget",
        "rationale"
})
public class FsPlanning {

    @JsonProperty("budget")
    private final FsBudget budget;

    @JsonProperty("rationale")
    private final String rationale;

    @JsonCreator
    public FsPlanning(@JsonProperty("budget") final FsBudget budget,
                      @JsonProperty("rationale") final String rationale) {
        this.budget = budget;
        this.rationale = rationale;
    }
}

File: src/main/kotlin/com/procurement/notice/model/budget/FsPlanning.kt
package com.procurement.notice.model.budget

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude

data class FsPlanning @JsonCreator constructor(

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val budget: FsBudget?,

    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val rationale: String?
)
